backend/hasura-actions/src/Query.res

```diff
@@ -36,19 +36,11 @@ module CloseStreamEntry = %graphql(`
 `)
 
 module UpdateStreamEntry = %graphql(`
-  mutation UpdateStreamEntry ($id: Int!, $paymentsMade: Int!, $paymentTick: Int!){
-    update_streams_by_pk(pk_columns: {id: $id}, _set: {numberOfPaymentsMade: $paymentsMade, paymentTick: $paymentTick}) {
+  mutation UpdateStreamEntry ($id: Int!, $paymentsMade: Int!, $nextPayment: Int!){
+    update_streams_by_pk(pk_columns: {id: $id}, _set: {numberOfPaymentsMade: $paymentsMade, nextPayment: $nextPayment}) {
       id
       numberOfPaymentsMade
-      paymentTick
-    }
-  }
-`)
-
-module DeleteStreamEntry = %graphql(`
-  mutation DeleteStreamEntry ($id: Int!){
-    delete_streams_by_pk(id: $id) {
-      id
+      nextPayment
```

Comment:
I've been thinking about this, we need to have new table in the database for payments, and this can have a `nullable` field called `currentPayment` or `mostRecentPayment` that references that payment.

Then the new payment table can have a column called `paymentState` that can start as "pending", go to "submitted" (after it is sent to the raiden node), and then either to "complete" (when the payment is complete) or "error" (if the payment fails).

Then in the `GetStreamData` you can fetch the most recent previous payment (if it exists), and check - if it is "complete" you just continue as before, if it has an "error" or is "pending" you can chose what to do. I'd say if it is pending, you can just skip it and wait until it isn't pending, if it is an "error" you can do the accounting to make the next payment a catch-up payment that incluedes the value of the previous payment.